core/modules/color/js/color.js

```diff
@@ -64,15 +65,20 @@ Backdrop.behaviors.color = {
           }
         }
         updatePreview();
+        $nameField.hide();
       }
     });
 
     $('input[data-color-name]').on('change', function () {
       var schemeName =  document.getElementById('edit-scheme').value;
       var key = this.dataset.colorName;
       if (schemeName !== '' && this.value !== schemes[schemeName][key]) {
+        $nameField.show();
```

Comment:
I'm trying to understand what is happening here 🤔 ...aren't we already handling this via `#states`, and if not, then shouldn't we?

Reply:
The states code should have been removed, which is done. We cant use states because we also need to toggle the custom name field when a user fiddles with the color buttons, which requires some client-side conditions.